framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2564,11 +2564,14 @@ protected ParameterizedExecutableType methodFromUse(
         // Store varargType before calling setParameterTypes, otherwise we may lose the varargType
         // as it is the last element of the original parameterTypes.
         method.computeVarargType();
-        // Adapt parameters, which makes parameters and arguments be the same size for later
-        // checking.
-        List<AnnotatedTypeMirror> parameters =
-                AnnotatedTypes.adaptParameters(this, method, tree.getArguments(), null);
-        method.setParameterTypes(parameters);
+
+        if (inferTypeArgs) {
```

Comment:
Let's do this now, but please file an issue that we should clean this up - either document when the varargs are expanded, change to the typetools way where the visitor has to do the adaptation, or do something nicer in the inference, where it shouldn't have to do adaptation again (it seems strange to do it there).